x-pack/plugins/security/public/management/users/edit_user/create_user_page.tsx

```diff
@@ -7,17 +7,25 @@
 
 import { EuiPageHeader, EuiSpacer } from '@elastic/eui';
 import type { FunctionComponent } from 'react';
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useHistory } from 'react-router-dom';
 
 import { FormattedMessage } from '@kbn/i18n-react';
 
+import { useCapabilities } from '../../../components/use_capabilities';
 import { UserForm } from './user_form';
 
 export const CreateUserPage: FunctionComponent = () => {
   const history = useHistory();
+  const readOnly = !useCapabilities('users').save;
   const backToUsers = () => history.push('/');
 
+  useEffect(() => {
+    if (readOnly) {
+      backToUsers();
+    }
+  }, [readOnly]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
I'm not sure I fully understand the use of `useEffect` here. As I understand it, it would trigger if `readOnly` updates. But can the `readOnly` boolean ever update being a simple type? I'm sure it's just my React understanding that's lacking 🤷‍♂️

Reply:
This effect will fire when the component mounts and when the readOnly prop changes. 

In both cases we want to redirect back to the index page if readOnly prop is true